src/http_server.rs

```diff
@@ -157,17 +170,37 @@ pub async fn run_server(
     addr: &SocketAddr,
     config: Arc<Config>,
 ) -> anyhow::Result<()> {
+    let cors = CorsLayer::new()
+        .allow_methods([Method::GET, Method::POST])
+        .allow_origin(Any);
+
     tracing::info!("Starting HTTP server listening on {addr}");
     let app = Router::new()
+
+        // TODO More convenient path would be `/datasets/:dataset/height`.
+        // Maybe it's better to preserve old URLs, but create new ones in new format
+        //
+        // .route("/datasets/:dataset/state", post(get_dataset_state))
+        // .route("/datasets/:dataset/height", post(get_height))
+        // .route("/datasets/:dataset/stream", post(execute_stream_restricted))
+        // .route("/datasets/:dataset/stream/debug", post(execute_stream))
+
         .route("/network/:dataset/height", get(get_height))
         .route("/network/:dataset/stream", post(execute_stream_restricted))
         .route("/network/:dataset/debug", post(execute_stream))
+        .route("/network/:dataset/state", get(get_dataset_state))
+
+
         // for backward compatibility
         .route("/network/:dataset/:start_block/worker", get(get_worker))
         .route("/query/:dataset_id/:worker_id", post(execute_query))
         .route("/network/state", get(get_network_state))
+        // end backward compatibility routes
```

Comment:
Actually, `/network/state` is also not for compatibility :)

Reply:
Okay!